src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -270,17 +270,27 @@ float64_t CCrossValidation::evaluate_one_run()
 			CMachine* machine;
 			CFeatures* features;
 			CLabels* labels;
+			CEvaluation* evaluation_criterion;
 
 			if (get_global_parallel()->get_num_threads()==1)
+			{
 				machine=m_machine;
+				features=m_features;
+				evaluation_criterion=m_evaluation_criterion;
+			}
 			else
+			{
 				machine=(CMachine*)m_machine->clone();
```

Comment:
I wonder whether the features of the machine are cloned here? I guess they will be. Then they will be overwritten with the manual clone below. Maybe it is good to remove any features from the `CMachine` instance before cloning to avoid doing this twice. Alternatively, set the features, clone machine, and then extract them from there.
Requires some checking of the existing code